File: src/dp/_198_rob.rs
struct Solution;

impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let mut dp = vec![];
        dp[0] = 0;
        dp[1] = nums[0];
        for i in 2..=nums.len() {
            dp[i] = dp[i - 1].max(dp[i - 2] + nums[i - 1])
        }
        dp[nums.len()]
    }
}

File: hot-100/198_rob.go
package hot_100

func rob(nums []int) int {
	dp := make([]int, len(nums)+1)
	dp[0] = 0
	dp[1] = nums[0]
	for i := 2; i < len(nums)+1; i++ {
		dp[i] = max(dp[i-1], dp[i-2]) + nums[i-1]
	}
	return dp[len(nums)]
}
